src/screens/Nearby/state.ts

```diff
@@ -439,3 +434,28 @@ export function useDepartureData(
     loadInitialDepartures,
   };
 }
+
+function getPayloadLocation(
+  locationOrStopPlace: Location | StopPlace,
+): DepartureGroupsPayloadLocation {
+  // locationOrStopPlace is StopPlace
+  if ('latitude' in locationOrStopPlace) {
+    return {
+      id: locationOrStopPlace.id,
+      layer: 'venue',
+    };
+  }
+
+  // locationOrStopPlace is Location
+  if ('coordinates' in locationOrStopPlace) {
+    return locationOrStopPlace.resultType === 'search'
+      ? locationOrStopPlace
+      : {
+          layer: 'address',
+          coordinates: locationOrStopPlace.coordinates,
+        };
+  }
+
+  // Should not be possible, but for typing there needs to be a fallback
+  return {id: '', layer: 'venue'};
```

Comment:
Can't this be avoided by just using 
```
if ('latitude' in locationOrStopPlace) {
    ...
} else {
    ...  
}

Reply:
I though so as well, but TypeScript doesn't seem to realize that it `locationOrStopPlace` has the `Location` type in that case.

<img width="516" alt="Screenshot 2022-10-11 at 09 41 46" src="https://user-images.githubusercontent.com/1774972/195026196-455931e8-e7de-4d46-824a-b8b982f27f78.png">

Reply:
It's because `latitude` is optional in `StopPlace`. Check on `resultType` instead, which is only in `Location`.

Reply:
Aha 💡 Updated here: https://github.com/AtB-AS/mittatb-app/pull/2997